conf/nomad/minio.hcl

```diff
@@ -125,18 +125,37 @@ job "${service_name}" {
         change_mode = "noop"
         env         = true
         data        = <<EOF
-%{ if use_vault_provider }
-{{ with secret "${vault_kv_path}" }}
-MINIO_ACCESS_KEY="{{ .Data.data.${vault_kv_field_access_key} }}"
-MINIO_SECRET_KEY="{{ .Data.data.${vault_kv_field_secret_key} }}"
-{{ end }}
-%{ else }
-MINIO_ACCESS_KEY="${access_key}"
-MINIO_SECRET_KEY="${secret_key}"
-%{ endif }
+
+  %{ if use_vault_provider }
+    {{ with secret "${vault_kv_path}" }}
+      MINIO_ACCESS_KEY="{{ .Data.data.${vault_kv_field_access_key} }}"
+      MINIO_SECRET_KEY="{{ .Data.data.${vault_kv_field_secret_key} }}"
+    {{ end }}
+  %{ else }
+    MINIO_ACCESS_KEY="${access_key}"
+    MINIO_SECRET_KEY="${secret_key}"
+  %{ endif }
 ${ envs }
 EOF
       }
+  %{ if use_vault_kms }
+    template {
+      data = <<EOH
+        {{ with secret "${vault_kms_approle_kv}" }}
+          MINIO_KMS_VAULT_APPROLE_ID="{{ .Data.data.approle_id }}"
+          MINIO_KMS_VAULT_APPROLE_SECRET="{{ .Data.data.secret_id }}"
+          {{end}}
+          MINIO_KMS_VAULT_ENDPOINT=${vault_address}
+          MINIO_KMS_VAULT_KEY_NAME=${vault_kms_key_name}
+          MINIO_KMS_VAULT_AUTH_TYPE=approle
+          MINIO_KMS_AUTO_ENCRYPTION=on
+          MINIO_KMS_VAULT_DEPRECATION=off
+        EOH
+
+        destination = "secrets/kms"
```

Comment:
Same question as above

Reply:
It´s set as static for simplicity I believe.